tests/test_page_image.py

```
@@ -0,0 +1,51 @@
+
+def test_rejects_unsupported_mode():
+    """Only 4-byte RGBX is supported; 24-bit RGB silently produces a 0x0
+    CGImage on macOS, so reject it at the boundary."""
+    with pytest.raises(ValueError, match="mode"):
+        PageImage(width=2, height=2, stride=6, buffer=b"\x00" * 12, mode="RGB")
+
+
+def test_is_immutable():
+    page = PageImage(width=2, height=2, stride=8, buffer=b"\x00" * 16, mode="RGBX")
+
+    with pytest.raises(Exception):
```

Ruff (0.16.1)

[warning] Do not assert blind exception: `Exception`

(B017)